api/library/EventSource/Store.hs

```diff
@@ -125,6 +131,29 @@ foldSubAsync sub onEvent onError =
   async $ foldSub sub onEvent onError
 
 --------------------------------------------------------------------------------
+foldSubSaved :: (MonadIO m)
+        => Subscription
+        -> (SavedEvent -> m ())
+        -> (SomeException -> m ())
+        -> m ()
+foldSubSaved sub onEvent onError = loop
+  where
+    loop = do
+      res <- nextEvent sub
+      case res of
+        Left e -> onError e
+        Right a -> onEvent a >> loop
+
+--------------------------------------------------------------------------------
+foldSubSavedAsync ::
```

Comment:
Please add a short documentation note. Also type signature is not aligned properly.